streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -289,8 +289,6 @@ public synchronized Topology addSource(final String name,
      * Add a new source that consumes from topics matching the given pattern and forwards the records to child processor
      * and/or sink nodes.
      * The source will use the specified key and value deserializers.
-     * The provided de-/serializers will be used for all matched topics, so care should be taken to specify patterns for
-     * topics that share the same key-value data format.
```

Comment:
Is this warning incorrect, aside from referring to "patterns"?

Reply:
The warning is correct, included it in the doc.